homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,181 @@
+        # Set the right type for mypy
+        self.coordinator: Alpha2BaseCoordinator = self.coordinator
+        self.heatarea_id = heatarea_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the climate device."""
+        return self.coordinator.data[self.heatarea_id]["HEATAREA_NAME"]
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return float(self.coordinator.data[self.heatarea_id].get("T_TARGET_MIN", 0.0))
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return float(self.coordinator.data[self.heatarea_id].get("T_TARGET_MAX", 30.0))
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return float(self.coordinator.data[self.heatarea_id].get("T_ACTUAL", 0.0))
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return current hvac mode."""
+        if self.coordinator.get_cooling():
+            return HVAC_MODE_COOL
+        return HVAC_MODE_HEAT
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        await self.coordinator.async_set_cooling(hvac_mode == HVAC_MODE_COOL)
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation."""
+        if not self.coordinator.data[self.heatarea_id]["_HEATCTRL_STATE"]:
+            return CURRENT_HVAC_IDLE
+        if self.coordinator.get_cooling():
+            return CURRENT_HVAC_COOL
+        return CURRENT_HVAC_HEAT
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        return float(self.coordinator.data[self.heatarea_id].get("T_TARGET", 0.0))
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperatures."""
+        target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        if target_temperature is None:
+            return
+
+        try:
+            await self.coordinator.async_set_target_temperature(
+                self.coordinator.data[self.heatarea_id]["ID"], target_temperature
+            )
+            self.coordinator.data[self.heatarea_id]["T_TARGET"] = target_temperature
+        except aiohttp.web.HTTPException as http_err:
+            raise HomeAssistantError(
+                "Failed to set target temperature, communication error with alpha2 base"
+            ) from http_err
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode."""
+        if self.coordinator.data[self.heatarea_id]["HEATAREA_MODE"] == 1:
+            return PRESET_DAY
+        if self.coordinator.data[self.heatarea_id]["HEATAREA_MODE"] == 2:
+            return PRESET_NIGHT
+        return PRESET_AUTO
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new operation mode."""
+        heatarea_mode = 0
```

Comment:
```suggestion
        heat_area_mode = 0
```

Reply:
I renamed all occurrences of heatarea to heat_area where possible.